### PR TITLE
Native App Integration tests

### PR DESCRIPTION
**Pull request by sfc-gh-cgorrie**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
- Adds integration tests for Native Applications
- Allows setting `SNOWFLAKE_CONNECTIONS_INTEGRATION_HOST`, but also leaving it blank
- widened types from `dict[str, str]` to `dict[str, Any]` for cursors being compared in integration tests (can return e.g. numbers back)
- centralize test projects in `tests/test_data/projects`

### TODO
- [x] get template repository open-sourced
- [x] re-enable init test

### CODE REVIEW

File: tests/project/integration/app/setup.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+create application role app_public;
+create or alter versioned schema core;
+
+    create or replace procedure core.echo(inp varchar)
```
Comment (sfc-gh-bdufour) [inline tests/project/integration/app/setup.sql:4]: is the indentation here intentional?

File: tests_integration/test_nativeapp.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+def pushd(dir: Path):
+    cwd = os.getcwd()
+    os.chdir(dir)
+    try:
+        yield dir
+    finally:
+        os.chdir(cwd)
+
+
+@pytest.mark.skip
```
Comment (sfc-gh-bdufour) [inline tests_integration/test_nativeapp.py:31]: ?

File: tests/project/integration/app/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# README
+
+This is the app/README.md file.
```
Comment (sfc-gh-bdufour) [inline tests/project/integration/app/README.md:3]: nit: consider saying this is a simple native app used for integ tests or something like that

File: tests/project/integration/app/setup.sql (hunk 1 above)
Comment (sfc-gh-cgorrie) [reply tests/project/integration/app/setup.sql:4]: Yep, it's my preferred way of making the different sections in a large setup script easy to read (each schema is a section).

File: tests_integration/test_nativeapp.py (hunk 2 above)
Comment (sfc-gh-bdufour) [reply tests_integration/test_nativeapp.py:31]: Ah, thanks, that's what I was missing. It's a token issue.

**Review by sfc-gh-bdufour — APPROVED** [review]

File: src/snowcli/cli/project/util.py [hunk 4]
```diff
@@ -60,3 +61,15 @@ def first_set_env(*keys: str):
...
+def is_mount(path: Path):
+    """
+    A wrapper around Path.is_mount() that does not throw an exception
+    on Windows. Instead, we treat Windows as if it has no mount points.
+    On POSIX platforms, behaviour is identical to Path.is_mount().
+    """
+    try:
+        return path.is_mount()
+    except NotImplementedError:
+        return False
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/cli/project/util.py:75]: Why not assume positive scenario?

File: tests/project/integration/app/README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# README
```
Comment (sfc-gh-turbaszek) [inline tests/project/integration/app/README.md:1]: We have project already under `tests/test_data/projects`

File: src/snowcli/cli/project/util.py (hunk 4 above)
Comment (sfc-gh-cgorrie) [reply src/snowcli/cli/project/util.py:75]: It doesn't really matter for our use case; we just use it as a stopping condition if it changes from the value of the initial directory. If you'd rather it default to `True` when not implemented, I'm fine with that.

I looked quickly to see if I could find the source code to understand how the Windows one works but my quick timebox didn't lead to the sourcecode :)

File: tests/project/integration/app/README.md (hunk 5 above)
Comment (sfc-gh-cgorrie) [reply tests/project/integration/app/README.md:1]: Hmm, I had put projects in this directory many weeks ago (for the initial project def implementation), is this the new home? If this is where you'd rather have all of them, would you like them moved in this PR?

Comment (sfc-gh-turbaszek) [reply tests/project/integration/app/README.md:1]: Yes, please let's move all projects under `tests_data`, we are using this place for long time. Sorry we haven't spotted this earlier

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: tests_integration/test_nativeapp.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    temp_dir,
+):
+    project_name = "myapp"
+    result = runner.invoke_with_config(
+        ["app", "init", project_name],
+        env=TEST_ENV,
+    )
+    assert result.exit_code == 0
+
+    with pushd(Path(os.getcwd(), project_name)):
```
Comment (sfc-gh-astus) [inline tests_integration/test_nativeapp.py:44]: @sfc-gh-turbaszek introduced project_file context manager https://github.com/Snowflake-Labs/snowcli/commit/3b59d1cb8f6b03ba5358d65ca21e7f2fc8623d44#diff-ad6b0ce118624337f6faf4e75af2aac6cc563a8632cbd95e699a04c1786f185dR208-R216

Logic is almost the same, but we will be consistent

Comment (sfc-gh-cgorrie) [reply tests_integration/test_nativeapp.py:44]: Do you mind if we refactor in a different ticket? This isn't the only place it's used (also in non-integration tests for definitions).

Comment (sfc-gh-astus) [reply tests_integration/test_nativeapp.py:44]: ok

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**